ui/apps/dashboard/src/layout/index.tsx

```diff
@@ -11,17 +11,39 @@ const { Sider: AntdSider, Content: AntdContent } = AntdLayout;
 
 export const MainLayout: FC = () => {
   const { authenticated } = useAuth();
+  const [collapsed, setCollapsed] = useState(false);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setCollapsed(window.innerWidth <= 768);
+    };
+
+    handleResize(); // Set initial state based on current window size
+    window.addEventListener('resize', handleResize);
+
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
   if (!authenticated) {
     return <Navigate to="/login" />;
   }
+
   return (
     <>
       <Header />
-      <AntdLayout className={cn('h-[calc(100vh-48px)]', 'overflow-hidden')}>
-        <AntdSider width={getSidebarWidth()}>
+      <AntdLayout className={cn('h-[calc(100vh-48px)]', 'overflow-hidden', 'flex')}>
+        <AntdSider
+          width={getSidebarWidth()}
+          collapsible
+          collapsed={collapsed}
+          breakpoint="lg"
+          onBreakpoint={(broken) => setCollapsed(broken)}
+          style={{ transition: 'width 0.3s ease' }}
```

Comment:
the antd lib already add the animation class, so it's no need to add duplicate style.
![image](https://github.com/user-attachments/assets/db378c89-7bc9-4a28-b2ae-202a873dcc88)